driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -341,6 +342,8 @@ private void write(Connection connection, Connection.ResponseCallback responseCa
                     break;
             }
 
+            // set this flag to true before writing the request
+            requestSent = true;
```

Comment:
Actually, exceptions thrown directly from `Connection#write` are caught in `query` which ignores the host and tries the next one directly, the retry policy is not involved there. So the only case where `requestSent` could be `false` would be from inside `writeHandler`.